common/src/Microsoft.Azure.IIoT.Core/src/Diagnostics/MetricLogger.cs

```diff
@@ -0,0 +1,48 @@
+// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.IIoT.Services.Diagnostics {
+    using Microsoft.ApplicationInsights;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.Azure.IIoT.Diagnostics;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Metric logger
+    /// </summary>
+    public sealed class MetricLogger : IMetricLogger  {
+
+        /// <summary>
+        /// Create metric logger
+        /// </summary>
+        /// <param name="config"></param>
+        public MetricLogger(IMetricLoggerConfig config) {
+            telemetryConfiguration = TelemetryConfiguration.Active;
+            telemetryConfiguration.InstrumentationKey = config.ApplicationInsightsInstrumentationKey;
```

Comment:
Ideally you make a copy of the active configuration before changing it since Active is a static global, so if it is used by everyone you are updating the instrumentation key and that could present timing issues, etc.   Could you check whether you can "Clone()" the active configuration into a member and then update it with the instrumentation key?